llama_index/response_synthesizers/google/generativeai/base.py

```diff
@@ -0,0 +1,265 @@
+    _answer_style: Any
+    _safety_setting: List[Any]
+
+    def __init__(
+        self,
+        *,
+        temperature: float,
+        answer_style: Any,
+        safety_setting: List[Any],
+        **kwargs: Any,
+    ):
+        """Create a new Google AQA.
+
+        Prefer to use the factory `from_defaults` instead for type safety.
+        See `from_defaults` for more documentation.
+        """
+        try:
+            import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        super().__init__(
+            service_context=google_service_context,
+            output_cls=SynthesizedResponse,
+            **kwargs,
+        )
+
+        self._client = genaix.build_generative_service()
+        self._temperature = temperature
+        self._answer_style = answer_style
+        self._safety_setting = safety_setting
+
+    # Type safe factory that is only available if Google is installed.
+    @classmethod
+    def from_defaults(
+        cls,
+        temperature: float = 0.7,
+        answer_style: int = 1,
+        safety_setting: List["genai.SafetySetting"] = [],
+    ) -> "GoogleTextSynthesizer":
+        """Create a new Google AQA.
+
+        Example:
+          responder = GoogleTextSynthesizer.create(
+              temperature=0.7,
+              answer_style=AnswerStyle.ABSTRACTIVE,
+              safety_setting=[
+                  SafetySetting(
+                      category=HARM_CATEGORY_SEXUALLY_EXPLICIT,
+                      threshold=HarmBlockThreshold.BLOCK_LOW_AND_ABOVE,
+                  ),
+              ]
+          )
+
+        Args:
+          temperature: 0.0 to 1.0.
+          answer_style: See `google.ai.generativelanguage.GenerateAnswerRequest.AnswerStyle`
+            The default is ABSTRACTIVE (1).
+          safety_setting: See `google.ai.generativelanguage.SafetySetting`.
+
+        Returns:
+          an instance of GoogleTextSynthesizer.
+        """
+        return cls(
+            temperature=temperature,
+            answer_style=answer_style,
+            safety_setting=safety_setting,
+        )
+
+    def get_response(
+        self,
+        query_str: str,
+        text_chunks: Sequence[str],
+        **response_kwargs: Any,
+    ) -> SynthesizedResponse:
+        """Generate a grounded response on provided passages.
+
+        Args:
+            query_str: The user's question.
+            text_chunks: A list of passages that should be used to answer the
+                question.
+
+        Returns:
+            A `SynthesizedResponse` object.
+        """
+        try:
+            import google.ai.generativelanguage as genai
+
+            import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        _logger.debug(
+            f"""\n
+GoogleTextSynthesizer.get_response(
```

Comment:
nit: the realignment is bit off

Reply:
Ooops. Don't need that debugging anymore! Removed. Thanks for catching this!